client/state/account-recovery/settings/reducer.js

```diff
@@ -1,33 +1,137 @@
 } from 'state/action-types';
 
-const reducer = createReducer( {}, {
-	[ ACCOUNT_RECOVERY_SETTINGS_FETCH ]: ( state ) => ( {
+const isUpdating = createReducer( {}, {
+	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE ]: ( state, { target } ) => ( {
 		...state,
-		isFetching: true,
+		[ target ]: true,
 	} ),
-
-	[ ACCOUNT_RECOVERY_SETTINGS_FETCH_SUCCESS ]: ( state, { email, email_validated, phone, phone_validated } ) => ( {
+	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_SUCCESS ]: ( state, { target } ) => ( {
 		...state,
-		email,
-		emailValidated: email_validated,
-		phone,
-		phoneValidated: phone_validated,
-		isFetching: false,
+		[ target ]: false,
 	} ),
+	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_FAILED ]: ( state, { target } ) => ( {
+		...state,
+		[ target ]: false,
+	} ),
+} );
 
-	[ ACCOUNT_RECOVERY_SETTINGS_FETCH_FAILED ]: ( state ) => ( {
+const isDeleting = createReducer( {}, {
+	[ ACCOUNT_RECOVERY_SETTINGS_DELETE ]: ( state, { target } ) => ( {
 		...state,
-		isFetching: false,
+		[ target ]: true,
+	} ),
+	[ ACCOUNT_RECOVERY_SETTINGS_DELETE_SUCCESS ]: ( state, { target } ) => ( {
+		...state,
+		[ target ]: false,
+	} ),
+	[ ACCOUNT_RECOVERY_SETTINGS_DELETE_FAILED ]: ( state, { target } ) => ( {
+		...state,
+		[ target ]: false,
 	} ),
 } );
 
-export default reducer;
+const convertPhoneResponse = ( phoneResponse ) => {
+	if ( ! phoneResponse ) {
+		return null;
+	}
+
+	const {
+		country_code,
+		country_numeric_code,
+		number,
+		number_full,
+	} = phoneResponse;
+
+	return {
+		countryCode: country_code,
+		countryNumericCode: country_numeric_code,
+		number: number,
+		numberFull: number_full,
+	};
+};
+
+const convertEmailResponse = ( emailResponse ) => {
+	return emailResponse ? emailResponse : '';
+};
+
+const data = createReducer( null, {
+	[ ACCOUNT_RECOVERY_SETTINGS_FETCH_SUCCESS ]: ( state, { settings } ) => {
+		const {
+			email,
+			email_validated,
+			phone,
+			phone_validated,
+		} = settings;
+
+		// At the moment, the response from /me/account-recovery could have an object or false for the phone field,
+		// and an string or false for the email field. I personally don't like the mixed value so did the conversion
+		// in the following.
+		return {
+			...state,
+			email: convertEmailResponse( email ),
+			emailValidated: email_validated,
+			phone: convertPhoneResponse( phone ),
+			phoneValidated: phone_validated,
+		};
+	},
+
+	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_SUCCESS ]: ( state, { target, value } ) => {
+		switch ( target ) {
+			case 'phone':
+				return {
+					...state,
+					phone: convertPhoneResponse( value ),
+				};
+			case 'email':
+				return {
+					...state,
+					email: convertEmailResponse( value ),
+				};
+			default: // do nothing to unknown targets
+				return { ...state };
+		}
+	},
+
+	[ ACCOUNT_RECOVERY_SETTINGS_DELETE_SUCCESS ]: ( state, { target } ) => {
+		switch ( target ) {
+			case 'phone':
+				return {
+					...state,
+					phone: null,
+				};
+			case 'email':
+				return {
+					...state,
```

Comment:
`data` is a good candidate to use another level of `combineReducers` to ensure that the state tree is initialized properly, especially since you already know what all the possible properties will be. The more manual object manipulation, the harder it can be to track down strange bugs where objects don't match what you expect. `combineReducers` helps ensure a fixed state tree shape to avoid those situations.

Eg:

```js
const phone = createReducer( null, {
	[ ACCOUNT_RECOVERY_SETTINGS_FETCH_SUCCESS ]: ( state, { settings } ) => {
		return convertPhoneResponse( settings.phone )
	},
	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_SUCCESS ]: ( state, { target, value } ) => {
		if ( target === 'phone' ) {
			return convertPhoneResponse( settings.phone );
		}
		return state;
	},
	[ ACCOUNT_RECOVERY_SETTINGS_DELETE_SUCCESS ]: ( state, { target, value } ) => {
		if ( target === 'phone' ) {
			return null;
		}
		return state;
	},
} );

// ...

export default combineReducers( {
	data: combineReducers( { phone, email } ),
	isUpdating,
	isDeleting,
} );
```

It would even be possible to take it a step further and build a function to generate the reducer:

```js
const createDataReducer = ( field, converter ) => {
	return createReducer( null, {
		[ ACCOUNT_RECOVERY_SETTINGS_FETCH_SUCCESS ]: ( state, { settings } ) => {
			return converter( settings[ field ] )
		},
		[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_SUCCESS ]: ( state, { target, value } ) => {
			if ( target === field ) {
				return converter( settings[ field ] );
			}
			return state;
		},
		[ ACCOUNT_RECOVERY_SETTINGS_DELETE_SUCCESS ]: ( state, { target, value } ) => {
			if ( target === field ) {
				return null;
			}
			return state;
		},
	} );
}

// ...

export default combineReducers( {
	data: combineReducers( {
		phone: createDataReducer( 'phone', convertPhoneResponse ),
		email: createDataReducer( 'email', convertEmailResponse ),
	} ),
	isUpdating,
	isDeleting,
} );
```